### PR TITLE
docs: nodeselector,affinities and topology considerations for requirements

### PR DESCRIPTION
**Pull request by sadath-12**
Fixes #N/A 

**Description**
While looking at the test cases it seems like the requirements also are considered with pod.spec.topologySpreadConstraints,  pod.spec.affinity.podAffinity, and pod.spec.nodeSelector rules. and we should mention in them docs if we are mentioning affinity too

**How was this change tested?**

**Does this change impact docs?**
- [x] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/docs/concepts/provisioners.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ spec:
     example.com/owner: "my-team"
 
   # Requirements that constrain the parameters of provisioned nodes.
-  # These requirements are combined with pod.spec.affinity.nodeAffinity rules.
+  # These requirements are combined with pod.spec.affinity.nodeAffinity and pod.spec.nodeSelector rules.
```
Comment (jonathan-innis) [inline website/content/en/docs/concepts/provisioners.md:61]:
```suggestion
  # These requirements are combined with pod.spec.topologySpreadConstraints, pod.spec.affinity.nodeAffinity, pod.spec.affinity.podAffinity, and pod.spec.nodeSelector rules.
```

I figure if we're touching this, we might as well add all the places that requirements come from

**Review by jonathan-innis — COMMENTED** [review]
Nice! I think that we can propagate this change to all of the docs

File: website/content/en/docs/concepts/provisioners.md (hunk 1 above)
Comment (sadath-12) [reply website/content/en/docs/concepts/provisioners.md]: ya makes sense

Comment (sadath-12) [reply website/content/en/docs/concepts/provisioners.md]: ya makes sense

**Comment by sadath-12** [comment]
Applied @jonathan-innis

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**